rest/api.go

```diff
@@ -332,13 +334,16 @@ func (h *handler) handleHeapProfiling() error {
 		return err
 	}
 	err = pprof.WriteHeapProfile(f)
+
+	fileCloseError := f.Close()
 	if err != nil {
-		return err
+		base.WarnfCtx(h.db.Ctx, "Error closing profile file: %v", fileCloseError)
 	}
-	err = f.Close()
+
```

Comment:
Can replace lines 343 to 347 with 'return err'

Reply:
Changes are committed to the latest revision.